test/jdk/java/awt/im/4959409/bug4959409.java

```diff
@@ -22,51 +22,199 @@
+            container.add(jTextField, BorderLayout.SOUTH);
+            container.add(jLabel, BorderLayout.CENTER);
+            frame.setSize(300, 300);
+            frame.setLocationRelativeTo(null);
+            frame.setAlwaysOnTop(true);
+            frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+            frame.setVisible(true);
+        });
+
+        if (!isTopLevelVisible(frame, frameVisibleLatch)) {
+            throw new RuntimeException("Dialog is not visible after " + TIMEOUT + "  seconds");
+        }
+        System.out.println("frame is visible " + frame.toString());
+
+        if (!isJComponentVisible(jTextField, componentVisibleLatch)) {
+            throw new RuntimeException("Component is not visible after " + TIMEOUT + "   seconds");
+        }
+
+        Robot robot = new Robot();
+        robot.setAutoDelay(DELAY);
+        robot.waitForIdle();
+
+        SwingUtilities.invokeAndWait(() -> {
+            points[0] = jTextField.getLocationOnScreen();
+            rect[0] = jTextField.getBounds();
+        });
+
+        performMouseAction(robot, points[0].x + rect[0].width / 2, points[0].y + rect[0].height / 2);
+
+        // Press SHIFT + 1 keys
+        robot.waitForIdle();
+        robot.keyPress(KeyEvent.VK_SHIFT);
+        robot.keyPress(KeyEvent.VK_1);
+        robot.keyRelease(KeyEvent.VK_1);
+        robot.keyRelease(KeyEvent.VK_SHIFT);
+        robot.waitForIdle();
+
+        if (!keyPressedEventLatch.await(TIMEOUT, TimeUnit.SECONDS)) {
+            throw new RuntimeException("Did not received KEY PRESS for Shift + 1 , test failed");
+        }
     }
-}
 
-class TestFrame extends JFrame implements KeyListener {
-    JTextField text;
-    JLabel label;
-
-    TestFrame () {
-        text = new JTextField();
-        text.addKeyListener(this);
-        label = new JLabel(" ");
-        Container c = getContentPane();
-        BorderLayout borderLayout1 = new BorderLayout();
-        c.setLayout(borderLayout1);
-        c.add(text, BorderLayout.CENTER);
-        c.add(label, BorderLayout.SOUTH);
-        setSize(300, 200);
-        setVisible(true);
+    public static void performMouseAction(final Robot robot, final int X, final int Y) {
+        robot.waitForIdle();
+        robot.mouseMove(X, Y);
+        robot.delay(DELAY);
+        robot.waitForIdle();
+
+        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+        robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+        robot.waitForIdle();
     }
 
-    public void keyPressed(KeyEvent e) {
-        int code = e.getKeyCode();
-        int mods = e.getModifiers();
-        if (code == '1' && mods == KeyEvent.SHIFT_MASK) {
-            label.setText("KEYPRESS received for Shift+1");
-        } else {
-            label.setText(" ");
+    public static void checkSwingTopLevelVisible(javax.swing.JFrame jFrame, CountDownLatch topLevelVisibleLatch) throws InterruptedException, InvocationTargetException {
```

Comment:
Is there a reason why you use the fully qualified `javax.swing.JFrame` when it's imported?

I suggest wrapping `throws` declaration to next line; this line is longer than 100.

Both comments apply to the method below.

Reply:
done